### ISSUE TITLE
Show Current Location on Map

### ISSUE THREAD

**Issue by metroshica** [bpoore/pbm-react#205]
While I love being able to look at my current location on the pinball map, sometimes it can be confusing where a pinball site is relative to my location if I don't know where I actually am. Obviously, this isn't a problem with a city I know well but I ran into this last weekend while on a road trip. I had to keep switching back and forth between Google maps and the pinball map to get an idea of how close I was to a location, since I wasn't familiar with the area. I'm wondering if it'd be possible to draw on the map where the user is, similar to Google maps. This is what I see when I open Google maps now.
![Screenshot_20190530-155429-resize](https://user-images.githubusercontent.com/7684916/58667630-cc6efb80-82f3-11e9-8d54-129347e8d1b7.png)

 
Thoughts?

**Comment by RyanTG** [comment]
Good idea!

**Comment by bpoore** [comment]
I agree, good idea! I believe it should be easy to get this added in.

**Comment by bpoore** [comment]
Here's a branch with this functionality. It looks good when not zoomed out too far, but the blue icon is easily covered up by map locations when zoomed out. I'm not quickly seeing a way to make the zIndex higher for the user's location icon than map locations.

https://github.com/bpoore/pbm-react/compare/current-location-icon?expand=1

**Comment by RyanTG** [comment]
Hmm yeah. Strangely, even if you move that MapView.Marker above the other one in the code, it still shows up below the red dots on the map.

**Comment by RyanTG** [comment]
I’ll look more into z index stuff. I could also change the border color of that icon do it doesn’t blend in as much when zoomed out.

**Comment by bpoore** [comment]
Yeah, it's weird... I tried moving the location of the marker to before and after the other markers and looked into adding zIndex styles, but couldn't get anything to work as of yet.

**Comment by RyanTG** [comment]
https://github.com/bpoore/pbm-react/pull/207
Who knew? There's a built-in `ShowUserLocation` function. I think it handles cases when the user hasn't granted the permission or doesn't have location services enabled. 

This animated is way better than the icon I made. And I spent 30 minutes trying to animate it before realizing the functionality is already there!

![mylocation-2](https://user-images.githubusercontent.com/632682/58684427-12948100-832d-11e9-8859-3df30a60ba20.gif)
